File: src/test/java/ru/yandex/praktikum/pages/HomePage.java
package ru.yandex.praktikum.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class HomePage {
    private WebDriver driver;
    private WebDriverWait wait;

    // Локаторы
    private final By cookieButton = By.id("rcc-confirm-button");
    private final By orderButtonTop = By.className("Button_Button__ra12g");
    private final By orderButtonBottom = By.xpath("//div[contains(@class, 'Home_FinishButton__1_cWm')]/button");
    private final By faqQuestions = By.className("accordion__button");
    private final By logoSamokat = By.className("Header_LogoScooter__3lsAR");
    private final By logoYandex = By.className("Header_LogoYandex__3TSOI");

    public HomePage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public void open() {
        driver.get("https://qa-scooter.praktikum-services.ru/");
    }

    public void acceptCookies() {
        try {
            WebElement cookie = driver.findElement(cookieButton);
            if (cookie.isDisplayed()) {
                cookie.click();
            }
        } catch (Exception e) {
            // Куки уже приняты
        }
    }

    public void clickOrderButtonTop() {
        wait.until(ExpectedConditions.elementToBeClickable(orderButtonTop)).click();
    }

    public void clickOrderButtonBottom() {
        WebElement button = driver.findElement(orderButtonBottom);
        ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true);", button);
        wait.until(ExpectedConditions.elementToBeClickable(button)).click();
    }

    public void clickFAQQuestion(int index) throws Exception {
        // Ждем загрузки всех вопросов
        wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(faqQuestions));
        WebElement question = driver.findElements(faqQuestions).get(index);
        // Скроллим к элементу по центру
        ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView({block: 'center'});", question);
        Thread.sleep(500);
        // Кликаем через JavaScript
        ((JavascriptExecutor) driver).executeScript("arguments[0].click();", question);
    }

    public String getFAQAnswer(int index) {
        By answerLocator = By.xpath("//div[@id='accordion__panel-" + index + "']/p");
        WebElement answer = wait.until(ExpectedConditions.visibilityOfElementLocated(answerLocator));
        return answer.getText();
    }

    public void clickSamokatLogo() {
        wait.until(ExpectedConditions.elementToBeClickable(logoSamokat)).click();
    }

    public void clickYandexLogo() {
        wait.until(ExpectedConditions.elementToBeClickable(logoYandex)).click();
    }
}

File: pages/home_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from pages.base_page import BasePage
import time

class HomePage(BasePage):
    # Локаторы
    COOKIE_BUTTON = (By.ID, "rcc-confirm-button")
    ORDER_BUTTON_TOP = (By.CLASS_NAME, "Button_Button__ra12g")
    ORDER_BUTTON_BOTTOM = (By.XPATH, "//div[contains(@class, 'Home_FinishButton__1_cWm')]/button")
    FAQ_QUESTIONS = (By.CLASS_NAME, "accordion__button")
    LOGO_SAMOKAT = (By.CLASS_NAME, "Header_LogoScooter__3lsAR")
    LOGO_YANDEX = (By.CLASS_NAME, "Header_LogoYandex__3TSOI")

    def open(self):
        self.driver.get("https://qa-scooter.praktikum-services.ru/")

    def accept_cookies(self):
        try:
            cookie_button = self.driver.find_element(*self.COOKIE_BUTTON)
            if cookie_button.is_displayed():
                cookie_button.click()
        except:
            pass

    def click_order_button_top(self):
        self.click_element(self.ORDER_BUTTON_TOP)

    def click_order_button_bottom(self):
        button = self.driver.find_element(*self.ORDER_BUTTON_BOTTOM)
        self.scroll_to_element(button)
        self.wait.until(EC.element_to_be_clickable(self.ORDER_BUTTON_BOTTOM))
        button.click()

    def click_faq_question(self, index):
        questions = self.find_elements(self.FAQ_QUESTIONS)
        question = questions[index]
        self.scroll_to_element(question)
        time.sleep(0.5)
        self.driver.execute_script("arguments[0].click();", question)

    def get_faq_answer(self, index):
        answer_locator = (By.XPATH, f"//div[@id='accordion__panel-{index}']/p")
        answer = self.wait.until(EC.visibility_of_element_located(answer_locator))
        return answer.text

    def click_samokat_logo(self):
        self.click_element(self.LOGO_SAMOKAT)

    def click_yandex_logo(self):
        self.click_element(self.LOGO_YANDEX)
